lib/src/core/optional.dart

```diff
@@ -26,35 +26,35 @@ class Optional<T> extends IterableBase<T> {
   /// Constructs an Optional of the given [value].
   ///
   /// Throws [ArgumentError] if [value] is null.
-  Optional.of(T value) : _value = value {
+  Optional.of(T /*!*/ value) : _value = value {
     if (_value == null) throw ArgumentError('Must not be null.');
```

Comment:
Null checks on T! aren't allowed in null safe mode, because they're not necessary. But, they _are_ needed in mixed mode. The recommended way is to use

https://api.flutter.dev/flutter/dart-core/ArgumentError/checkNotNull.html

Reply:
While `T!` is not allowed, `T/*!*/` is used by the migration tool as a signal that means "do not make this nullable", and is converted to just `T`.